src/nvim/options.lua

```diff
@@ -2567,7 +2567,7 @@ return {
       vi_def=true,
       expand=true,
       varname='p_vdir',
-      defaults={if_true={vi=macros('DFLT_VDIR')}}
+      defaults={if_true={vi=''}}
```

Comment:
I see that only `viewdir` is touched. Are there plans for other options from #78? Especially for &runtimepath.

Don’t forget that for default &runtimepath value you may take code from my `pr-3120` branch:

```Patch
diff --git a/src/nvim/option.c b/src/nvim/option.c
index 9a375c0..f9b8924 100644
--- a/src/nvim/option.c
+++ b/src/nvim/option.c
@@ -301,6 +301,148 @@ static char *(p_cot_values[]) = {"menu", "menuone", "longest", "preview",
 # include "option.c.generated.h"
 #endif
 
+/// Count commas in the given string
+static size_t count_commas(const char *const s, size_t len)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  size_t ret = 0;
+  for (size_t i = 0; i < len; i++) {
+    if (s[i] == ',') {
+      ret++;
+    }
+  }
+  return ret;
+}
+
+/// Append string with escaped commas
+static char *strcpy_comma_escaped(char *dest, const char *src, const size_t len)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  size_t shift = 0;
+  for (size_t i = 0; i < len; i++) {
+    if (src[i] == ',') {
+      dest[i + shift++] = '\\';
+    }
+    dest[i + shift] = src[i];
+  }
+  return &dest[len + shift];
+}
+
+/// Set &runtimepath to default value
+static void set_runtimepath_default(void)
+{
+  size_t rtp_size = 0;
+  char *const data_home = vim_getenv("XDG_DATA_HOME");
+  char *const config_home = vim_getenv("XDG_CONFIG_HOME");
+  char *const vimruntime = vim_getenv("VIMRUNTIME");
+  char *const data_dirs = vim_getenv("XDG_DATA_DIRS");
+  char *const config_dirs = vim_getenv("XDG_CONFIG_DIRS");
+#define NVIM_SIZE (sizeof("/nvim") - 1)
+#define SITE_SIZE (sizeof("/site") - 1)
+#define AFTER_SIZE (sizeof("/after") - 1)
+  size_t data_len;
+  size_t config_len;
+  size_t vimruntime_len;
+  if (data_home != NULL) {
+    data_len = strlen(data_home);
+    rtp_size += ((data_len + count_commas(data_home, data_len)
+                  + NVIM_SIZE + SITE_SIZE) * 2 + AFTER_SIZE) + 2;
+  }
+  if (config_home != NULL) {
+    config_len = strlen(config_home);
+    rtp_size += ((config_len + count_commas(config_home, config_len)
+                  + NVIM_SIZE) * 2 + AFTER_SIZE) + 2;
+  }
+  if (vimruntime != NULL) {
+    vimruntime_len = strlen(vimruntime);
+    rtp_size += vimruntime_len + count_commas(vimruntime, vimruntime_len) + 1;
+  }
+#define COMPUTE_COLON_LEN(rtp_size, additional_size, val) \
+  do { \
+    if (val != NULL) { \
+      const void *iter = NULL; \
+      do { \
+        size_t dir_len; \
+        const char *dir; \
+        iter = vim_colon_env_iter(val, iter, &dir, &dir_len); \
+        if (dir != NULL && dir_len > 0) { \
+          rtp_size += ((dir_len + count_commas(dir, dir_len) \
+                        + NVIM_SIZE + additional_size) * 2 \
+                       + AFTER_SIZE) + 2; \
+        } \
+      } while (iter != NULL); \
+    } \
+  } while (0)
+  COMPUTE_COLON_LEN(rtp_size, SITE_SIZE, data_dirs);
+  COMPUTE_COLON_LEN(rtp_size, 0, config_dirs);
+#undef COMPUTE_COLON_LEN
+  if (rtp_size == 0) {
+    return;
+  }
+  // All additions were including comma.
+  rtp_size--;
+  char *const rtp = xmallocz(rtp_size);
+  char *rtp_cur = rtp;
+#define ADD_STRING(tgt, src, len) \
+  tgt = strcpy_comma_escaped(tgt, src, len)
+#define ADD_STATIC_STRING(tgt, src) \
+  do { memmove(tgt, src, sizeof(src) - 1); tgt += sizeof(src) - 1; } while (0)
+#define ADD_COLON_DIRS(tgt, val, suffix, revsuffix) \
+  do { \
+    if (val != NULL) { \
+      const void *iter = NULL; \
+      do { \
+        size_t dir_len; \
+        const char *dir; \
+        iter = vim_colon_env_iter##revsuffix(val, iter, &dir, &dir_len); \
+        if (dir != NULL && dir_len > 0) { \
+          ADD_STRING(rtp_cur, dir, dir_len); \
+          ADD_STATIC_STRING(rtp_cur, "/nvim" suffix ","); \
+        } \
+      } while (iter != NULL); \
+    } \
+  } while (0)
+  if (config_home != NULL) {
+    ADD_STRING(rtp_cur, config_home, config_len);
+    ADD_STATIC_STRING(rtp_cur, "/nvim,");
+  }
+  ADD_COLON_DIRS(rtp_cur, config_dirs, "", );
+  if (data_home != NULL) {
+    ADD_STRING(rtp_cur, data_home, data_len);
+    ADD_STATIC_STRING(rtp_cur, "/nvim/site,");
+  }
+  ADD_COLON_DIRS(rtp_cur, data_dirs, "/site", );
+  if (vimruntime != NULL) {
+    ADD_STRING(rtp_cur, vimruntime, vimruntime_len);
+    *rtp_cur++ = ',';
+  }
+  ADD_COLON_DIRS(rtp_cur, data_dirs, "/site/after", _rev);
+  if (data_home != NULL) {
+    ADD_STRING(rtp_cur, data_home, data_len);
+    ADD_STATIC_STRING(rtp_cur, "/nvim/site/after,");
+  }
+  ADD_COLON_DIRS(rtp_cur, config_dirs, "/after", _rev);
+  if (config_home != NULL) {
+    ADD_STRING(rtp_cur, config_home, config_len);
+    ADD_STATIC_STRING(rtp_cur, "/nvim/after");
+  } else {
+    // Strip trailing comma.
+    rtp[rtp_size] = NUL;
+  }
+#undef ADD_COLON_DIRS
+#undef ADD_STATIC_STRING
+#undef ADD_STRING
+#undef NVIM_SIZE
+#undef SITE_SIZE
+#undef AFTER_SIZE
+  set_string_default("runtimepath", (char_u *)rtp);
+  xfree(data_dirs);
+  xfree(config_dirs);
+  xfree(data_home);
+  xfree(config_home);
+  xfree(vimruntime);
+}
+
 /*
  * Initialize the options, first part.
  *
@@ -437,6 +579,10 @@ void set_init_1(void)
       "system('lpr' . (&printdevice == '' ? '' : ' -P' . &printdevice) . ' ' . v:fname_in) . delete(v:fname_in) + v:shell_error"
       );
 
+  // Set default for &runtimepath. All necessary expansions are performed in 
+  // this function.
+  set_runtimepath_default();
+
   /*
    * Set all the options (except the terminal options) to their default
    * value.  Also set the global value for local options.
diff --git a/src/nvim/os/env.c b/src/nvim/os/env.c
index 7be8a86..e47a63b 100644
--- a/src/nvim/os/env.c
+++ b/src/nvim/os/env.c
@@ -415,6 +415,74 @@ static char *remove_tail(char *p, char *pend, char *name)
   return pend;
 }
 
+/// Iterate over colon-separated list
+///
+/// @note Environment variables must not be modified during iteration.
+///
+/// @param[in]   val   Value of the environment variable to iterate over.
+/// @param[in]   iter  Pointer used for iteration. Must be NULL on first
+///                    iteration.
+/// @param[out]  dir   Location where pointer to the start of the current
+///                    directory name should be saved. May be set to NULL.
+/// @param[out]  len   Location where current directory length should be saved.
+///
+/// @return Next iter argument value or NULL when iteration should stop.
+const void *vim_colon_env_iter(const char *const val,
+                               const void *const iter,
+                               const char **const dir,
+                               size_t *const len)
+  FUNC_ATTR_NONNULL_ARG(1,3,4) FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  const char *varval = (const char *) iter;
+  if (varval == NULL) {
+    varval = val;
+  }
+  *dir = varval;
+  const char *const dirend = strchr(varval, ':');
+  if (dirend == NULL) {
+    *len = strlen(varval);
+    return NULL;
+  } else {
+    *len = (size_t) (dirend - varval);
+    return dirend + 1;
+  }
+}
+
+/// Iterate over colon-separated list in reverse order
+///
+/// @note Environment variables must not be modified during iteration.
+///
+/// @param[in]   val   Value of the environment variable to iterate over.
+/// @param[in]   iter  Pointer used for iteration. Must be NULL on first
+///                    iteration.
+/// @param[out]  dir   Location where pointer to the start of the current
+///                    directory name should be saved. May be set to NULL.
+/// @param[out]  len   Location where current directory length should be saved.
+///
+/// @return Next iter argument value or NULL when iteration should stop.
+const void *vim_colon_env_iter_rev(const char *const val,
+                                   const void *const iter,
+                                   const char **const dir,
+                                   size_t *const len)
+  FUNC_ATTR_NONNULL_ARG(1,3,4) FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  const char *varend = (const char *) iter;
+  if (varend == NULL) {
+    varend = val + strlen(val) - 1;
+  }
+  const size_t varlen = (size_t) (varend - val) + 1;
+  const char *const colon = xmemrchr(val, ':', varlen);
+  if (colon == NULL) {
+    *len = varlen;
+    *dir = val;
+    return NULL;
+  } else {
+    *dir = colon + 1;
+    *len = (size_t) (varend - colon);
+    return colon - 1;
+  }
+}
+
 /// Vim's version of getenv().
 /// Special handling of $HOME, $VIM and $VIMRUNTIME, allowing the user to
 /// override the vim runtime directory at runtime.  Also does ACP to 'enc'
```

, just replace `vim_getenv` calls.